cognitive/src/main/scala/com/microsoft/azure/synapse/ml/cognitive/AnomalyDetectionBlobHelpers.scala

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.cognitive
+
+import com.azure.storage.blob.sas.{BlobSasPermission, BlobServiceSasSignatureValues}
+import com.azure.storage.blob.{BlobContainerClient, BlobServiceClientBuilder}
+import com.azure.storage.common.StorageSharedKeyCredential
+import org.apache.spark.sql.types.StructField
+import org.apache.spark.sql.{DataFrame, Row}
+
+import java.io.{ByteArrayInputStream, ByteArrayOutputStream, OutputStream, PrintWriter}
+import java.time.format.DateTimeFormatter
+import java.time.{Instant, OffsetDateTime, ZoneId, ZoneOffset}
+import java.util.UUID
+import java.util.zip.{ZipEntry, ZipOutputStream}
+
+object AnomalyDetectionBlobHelpers {
+
+  def apply(storageConnectionString: String, containerName: String,
+            timestampCol: String): AnomalyDetectionBlobHelpers = {
+    val blobContainerClient = new BlobServiceClientBuilder()
+      .connectionString(storageConnectionString)
+      .credential(StorageSharedKeyCredential.fromConnectionString(storageConnectionString))
+      .buildClient()
+      .getBlobContainerClient(containerName.toLowerCase())
+    if (!blobContainerClient.exists()) {
+      blobContainerClient.create()
+    }
+    new AnomalyDetectionBlobHelpers(blobContainerClient, timestampCol)
+  }
+
+  def apply(storageName: String, storageKey: String, endpoint: String, sasToken: String,
+            containerName: String, timestampCol: String): AnomalyDetectionBlobHelpers = {
+    val blobContainerClient = new BlobServiceClientBuilder()
+      .endpoint(endpoint)
+      .sasToken(sasToken)
+      .credential(new StorageSharedKeyCredential(storageName, storageKey))
+      .buildClient()
+      .getBlobContainerClient(containerName.toLowerCase())
+    if (!blobContainerClient.exists()) {
+      blobContainerClient.create()
+    }
+    new AnomalyDetectionBlobHelpers(blobContainerClient, timestampCol)
+  }
+}
+
+/** Helper to upload data from dataframe to Azure Blob storage. Stored as CSV and zipped.
+  *
+  * @param blobContainerClient configured with credentials.
+  * @param timestampCol        timestamp column of the source dataframe.
+  */
+class AnomalyDetectionBlobHelpers(blobContainerClient: BlobContainerClient, timestampCol: String) {
+
+  var blobName = ""
```

Comment:
this mutable var worries me, seems like you could just return this from the upload function